platforms/defaults_windows.go

```diff
@@ -53,6 +55,31 @@ func (m windowsmatcher) Match(p specs.Platform) bool {
 		if strings.HasPrefix(p.OSVersion, m.osVersionPrefix) {
 			return true
 		}
+		if m.isClientOS && p.OSVersion != "" {
+			split := strings.Split(p.OSVersion, ".")
```

Comment:
We already indirectly import a package for version parsing ([go.mod](https://github.com/containerd/containerd/blob/1944ef13e218a979da3d4a5f2c6695bd11e1b02a/go.mod#LL87C19-L87C25), [vendor](https://github.com/containerd/containerd/tree/main/vendor/github.com/blang/semver/v4)). Should we just reuse it instead of maintaining a pretty fragile logic?

Reply:
Unfortunately, what this is parsing isn't semver (it's Windows 4-part OS versions like 10.0.20348.1850), so from what I can tell, that library won't actually parse them correctly. :sweat_smile: